packages/common/src/MigrationUtils.js

```diff
@@ -156,6 +164,38 @@ async function addToTdr(instance, network) {
   }
 }
 
+// "Tvr" - Truffle verification registry
+async function addToTvr(address, args, network, networkId) {
+  // Adds in constructor args for contracts in the k, v store
+  // e.g. Produces a networks/1_args.json
+  // with the structure
+  // { "address": "args" }
+  if (!tdr.isDryRunNetworkName(network) && shouldCommitDeployment(network)) {
+    const tvrPath = path.join(process.cwd(), "networks", `${networkId}_args.json`);
+
+    let tvrData = {};
+
+    // If file exists, just read it
+    if (fs.existsSync(tvrPath)) {
+      tvrData = JSON.parse(fs.readFileSync(tvrPath));
+    }
+
+    // Remove { 'from': ... } thats present in the deployment args
+    // As well as empty objects
+    // And destructure the single tuple (with rawValue keys)
+    const argsFixed = args
+      .filter(x => !x.from && (typeof x === "object" ? Object.keys(x).length > 0 : true))
+      .map(x => {
+        // Tuple
+        if (x.rawValue) return [x.rawValue];
+        return x;
+      });
+
+    // Save to file
+    fs.writeFileSync(tvrPath, JSON.stringify({ ...tvrData, [address]: argsFixed }, null, 4));
```

Comment:
Can you explain what the `null` and `4` args passed to JSON.stringfy do and why they're needed? Seems like a bit of a magic number

Reply:
The `null` and `4` are there to make it human readable, otherwise it'll flatten the whole thing.

The `null` is the `replacer` function, which intends to replace key-value pairs.

`4` is number of spacing. 4 was chosen as thats the standard number of indented spaces.